src/Microsoft.DotNet.Darc/DarcLib/Helpers/ManifestHelper.cs

```diff
@@ -164,21 +173,34 @@ private static List<DownloadedAsset> SelectMergedManifestAssets(IEnumerable<Down
                 // it is taken from the Build.Name, which corresponds to the repository name.
                 assetOrigin ??= repoName;
 
-                if (assetOriginMap.TryGetValue(assetName, out var existingAssetOrigin))
+                if (assetMetadataMap.TryGetValue(assetName, out var existingAssetMetadata))
                 {
-                    if (existingAssetOrigin != assetOrigin)
+                    if (existingAssetMetadata.Origin != assetOrigin)
                     {
                         logger.LogWarning($"Warning: The same asset '{assetName}' is listed in various '{MergedManifestFileName}', " +
-                            $"with differing origins specified in each: {existingAssetOrigin}, {assetOrigin}.");
+                            $"with differing origins specified in each: {existingAssetMetadata.Origin}, {assetOrigin}.");
                     }
                 }
                 else
                 {
-                    assetOriginMap.Add(assetName, assetOrigin);
+                    assetMetadataMap.Add(assetName, new AssetReleaseMetadata(assetOrigin, dotNetReleaseShipping));
                 }
             }
         }
 
-        return assetOriginMap;
+        return assetMetadataMap;
+    }
+
+    private class AssetReleaseMetadata
+    {
+        public AssetReleaseMetadata(string origin, bool dotNetReleaseShipping)
+        {
+            Origin = origin;
+            DotNetReleaseShipping = dotNetReleaseShipping;
+        }
+
+        public string Origin { get; set; }
```

Comment:
maybe we can make both of these `{ get; init; }` so they're immutable. Doesn't look like we're changing them after initialization

Reply:
I'll  use a `record`